crates/client/src/test_generator.rs

```diff
@@ -100,18 +100,15 @@ macro_rules! load_template {
 /// # Fields
 /// - `root`: A `PathBuf` indicating the root directory of the project for which tests are being generated.
 /// This path is used as a base for any relative paths within the project.
-/// - `idl`: An `Idl` struct. This field is used to understand the interfaces
-/// and data structures that tests may need to interact with.
-/// - `codes_libs_pairs`: A vector of tuples, each containing a `String` and a `Utf8PathBuf`.
+/// - `programs_data`: A vector of tuples, each containing a `String`, `Utf8PathBuf` and IDL Program data.
 /// Each tuple represents a pair of code and the package path associated with it.
 /// - `packages`: A vector of `Package` structs, representing the different packages
 /// that make up the project.
 /// - `use_tokens`: A vector of `ItemUse` tokens from the Rust syntax, representing `use` statements that
 /// should be included in the generated code for .program_client.
 pub struct TestGenerator {
     pub root: PathBuf,
-    pub idl: Idl,
-    pub codes_libs_pairs: Vec<(String, Utf8PathBuf)>,
+    pub programs_data: Vec<(String, Utf8PathBuf, IdlProgram)>,
```

Comment:
I don't like tuples too much, because it makes it hard to read the code. What does the String represents? And what is some_variable.2 ? Etc... What about having a new struct? Such as:
```rust
pub struct ProgramData {
    program_name: String,
    path: ...,
    idl: ...
}
```